Tasks/Common/VstsAzureHelpers_/InitializeFunctions.ps1

```diff
@@ -150,18 +169,49 @@ function Set-CurrentAzureRMSubscription {
     $null = Select-AzureRMSubscription -SubscriptionId $SubscriptionId @additional
 }
 
-function Format-Splat {
+function Set-UserAgent {
```

Comment:
Pasting my previous questions again (from previous comment on now-outdated diff):

* I really don't like this line at all:
`[Microsoft.Azure.Common.Authentication.AzureSession]::ClientFactory.AddUserAgent($UserAgent)`.
  * You're importing a module, and then making assumptions about the internal implementation details of that module. You don't control which version of the Azure cmdlets are on disk.
  * Does this work with all of the range of Azure/AzureRM module versions currently supported by this module?
  * Does this work for AzureRM and for classic Azure? What if only one of AzureRM or Azure is imported? What if both are imported? Both imported is a supported scenario for username/pw endpoint.
  * Do the Azure/AzureRM cmdlets expose a cmdlet to set this? If not, can we request one? It seems like we would want to prefer the cmdlet if one is available, and then fallback to this attempt for older versions of the module that don't expose the cmdlet.
  * Can you add a try/catch around the type initialization and write the exception details to the verbose stream if it fails?
  * If continuing down the type initialization route, I would prefer to see your code check whether type exists, and then check whether the expected method signature exists (`Get-Member` cmdlet can be used) before attempting to invoke the method.
* Can you add the following at the beginning of this function (see Initialize-Azure for an example):
```
[CmdletBinding()]
param()
```
* Can you move this down to be sorted alphabetically like the rest of the functions? Albeit `Format-Splat` is also out of order (at the bottom); feel free to fix that.
* Can you fix the formatting to be consistent with rest of the file?
* Can you change this so `Initialize-AzureSubscription` invokes `Set-UserAgent`, rather than from the psm1 file? Based on how this module is currently organized, that would be a better organizational fit.
* Can you change all variable references to use Get-VstsTaskVariable? Tracing is built-in to that lib funciton. E.g. change `$env:SYSTEM_TEAMFOUNDATIONCOLLECTIONURI` to `Get-VstsTaskVariable -Name System.TeamFoundationCollectionUri -Require`
* What if host type is not build or release? Wouldn't it be better to not add an empty user agent string? Can you change your branching structure to `if ... elseif ... else { write-verbose "some message that indicates it's not either" ; return }`.
* Please add L0 tests. I have put forth great effort to ensure this module has extensive L0 coverage. You will need to move the call to `[Microsoft.Azure.Common.Authentication.AzureSession]::ClientFactory.AddUserAgent($UserAgent)` behind a wrapper function, e.g.
```
function Set-UserAgent_Core {
    [CmdletBinding()]
    param(
        [Parameter(Mandatory = $true)]
        [string]$UserAgent)

    Trace-VstsEnteringInvocation $MyInvocation
    try {
        [Microsoft.Azure.Common.Authentication.AzureSession]::ClientFactory.AddUserAgent($UserAgent)
    } finally {
        Trace-VstsLeavingInvocation $MyInvocation
    }
}
```

...the `Trace-VstsEnteringInvocation` function will trace the parameter for you.